external-crates/move/crates/move-compiler/tests/move_2024/migration/all_migrations.exp

```diff
@@ -26,7 +26,7 @@ warning[W09003]: unused assignment
    ┌─ tests/move_2024/migration/all_migrations.move:12:20
    │
 12 │         let S { f: fin } = s;
-   │                    ^^^ Unused assignment for variable 'fin'. Consider removing, replacing with '_', or prefixing with '_' (e.g., '_fin')
+   │                    ^^^ Unused variable 'fin'. Consider removing, replacing with '_', or prefixing with '_' (e.g., '_fin')
```

Comment:
I believe this error needs to be nicer. This is an error from the following code, which should possibly live as its own test:

```
module a::m {
    struct S { f: u64 }

    fun t0(x: u64, s: S): u64 {
        let S { f: fin } = s;
        fin = 10;
        x = 20;
        fin + x
    }

}
```

I would expect mutability errors for both `fin` and `x`, but this error here is very strange. It's because the value for `fin` is dropped during reassignment, and it would be cool if we could get the error to explain that: renaming this as `_fin` is not really the correct fix.

Reply:
Fixed this, so its back to complaining about an unused assignment